### PR TITLE
Complete Grid Printer assignment

### PR DESCRIPTION
**Pull request by stevetwalker**
Let me know if you need me to include any error messages!

### CODE REVIEW

**Comment by stevetwalker** [comment]
Lesson 2: Fizz Buzz assignment

**Comment by stevetwalker** [comment]
Lesson 2: series.py

**Comment by stevetwalker** [comment]
Lesson 3: slicing_lab

**Comment by stevetwalker** [comment]
Lesson 3: list_lab.py

**Comment by stevetwalker** [comment]
Lesson 3: strformat_lab.py

File: students/steve_walker/lesson02/fizz_buzz.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+for i in range(1, 101):
+	if i % 15 == 0:
```
Comment (charlesrobison) [inline students/steve_walker/lesson02/fizz_buzz.py]: Over indented - only 4 spaces in Python

File: students/steve_walker/lesson02/fizz_buzz.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+		print("FizzBuzz")
+	elif i % 5 == 0:
+		print("Buzz")
+	elif i % 3 == 0:
+		print("Fizz")
+	else:
+		print(i)
```
Comment (charlesrobison) [inline students/steve_walker/lesson02/fizz_buzz.py]: Works well!

File: students/steve_walker/lesson02/grid_printer.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+def print_grid(x,y):
+	
+	# create the horizontal (cross) lines
+	segment = " -"
```
Comment (charlesrobison) [inline students/steve_walker/lesson02/grid_printer.py]: Over indented

File: students/steve_walker/lesson02/grid_printer.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	# print the grid
+	for i in range(y-1):
+		print(cross)
+		print(vertline)
+	print(cross)
+
+print_grid(9,5)
+
+# To the teacher: let me know if you'd like error messages for invalid
+# inputs (like negative numbers) or whether I can trust you to behave!
```
Comment (charlesrobison) [inline students/steve_walker/lesson02/grid_printer.py]: We'll get to error messages later - thanks for asking!

File: students/steve_walker/lesson02/series.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    # test if sum_series matched lucas
+    assert sum_series(5, 2, 1) == lucas(5)
+
+    # test if sum_series works for arbitrary initial values
+    assert sum_series(0, 3, 2) == 3
+    assert sum_series(1, 3, 2) == 2
+    assert sum_series(2, 3, 2) == 5
+    assert sum_series(3, 3, 2) == 7
+    assert sum_series(4, 3, 2) == 12
+    assert sum_series(5, 3, 2) == 19
```
Comment (charlesrobison) [inline students/steve_walker/lesson02/series.py]: Nice assert statements!

File: students/steve_walker/lesson03/list_lab.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+#Series 1
+fruitlist = ["Apples", "Pears", "Oranges", "Peaches"]
+print(fruitlist)
+
+add_fruit = input("\nAnother fruit, perhaps? What would you like?\n")
+fruitlist.append(add_fruit)
+print(fruitlist)
+
+number = int(input("\nPick a number from 1 to " + str(len(fruitlist)) + ".\n"))
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: Nice use of len()

File: students/steve_walker/lesson03/list_lab.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+print(number, ": ", fruitlist[number-1],"\n")
+
+fruitlist = ["Guava"] + fruitlist
+print(fruitlist)
+
+fruitlist.insert(0, "Mango")
+print(fruitlist)
+
+for fruit in fruitlist:
+	if fruit[0] == "P":
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: Over indented

File: students/steve_walker/lesson03/list_lab.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+print(fruitlist)
+
+add_fruit = input("\nAnother fruit, perhaps? What would you like?\n")
+fruitlist.append(add_fruit)
+print(fruitlist)
+
+number = int(input("\nPick a number from 1 to " + str(len(fruitlist)) + ".\n"))
+print(number, ": ", fruitlist[number-1],"\n")
+
+fruitlist = ["Guava"] + fruitlist
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: You'll want snake_case variables as well as function names in Python.  So, fruit_list.

File: students/steve_walker/lesson03/list_lab.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+#Series 2 bonus
+fruitlist2 = fruitlist2*2
+
+while True:
+	try:
+		xfruit = input("\nWhat fruit should we take away?\n")
+		fruitlist2.index(xfruit)
+		break
+	except:
+		print("Let's behave. Try another fruit!")
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: :)

File: students/steve_walker/lesson03/list_lab.py [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+print(fruitlist2)
+del fruitlist2[-1]
+print(fruitlist2)
+
+#Series 2 bonus
+fruitlist2 = fruitlist2*2
+
+while True:
+	try:
+		xfruit = input("\nWhat fruit should we take away?\n")
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: You can use upper() or the like here too to catch case sensitive input

File: students/steve_walker/lesson03/list_lab.py [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+fruitlist2 = fruitlist[:]
+print(fruitlist2)
+del fruitlist2[-1]
+print(fruitlist2)
+
+#Series 2 bonus
+fruitlist2 = fruitlist2*2
+
+while True:
+	try:
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: Over indented

File: students/steve_walker/lesson03/list_lab.py [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+		yn = input(f"Do you like {fruit.lower()}? (yes or no)\n")
+		if yn != "yes" and yn != "no":
+			print("Let's behave. Just type yes or no, please!\n")
+			
+		elif yn == "no":
+			fruitlist3 = [x for x in fruitlist3 if x != fruit]
+			break
+		
+		else:
+			break
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/list_lab.py]: Works well!

File: students/steve_walker/lesson03/slicing_lab.py [hunk 13]
```diff
@@ -0,0 +1,52 @@
+
+def exchange_first_last(seq):
+	return(seq[-1:] + seq[1:-1] + seq[:1])
```
Comment (charlesrobison) [inline students/steve_walker/lesson03/slicing_lab.py]: Over indented

**Pull request merged**